src/main/java/io/supertokens/webserver/api/oauth/OAuthAPI.java

```diff
@@ -0,0 +1,98 @@
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.oauth;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.supertokens.Main;
+import io.supertokens.httpRequest.HttpResponseException;
+import io.supertokens.oauth.OAuth;
+import io.supertokens.oauth.exceptions.OAuthException;
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.Storage;
+import io.supertokens.pluginInterface.exceptions.InvalidConfigException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.TenantIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.pluginInterface.oauth.OAuthAuthResponse;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.catalina.Store;
+
+import java.io.IOException;
+import java.io.Serial;
+
+public class OAuthAPI extends WebserverAPI {
+    @Serial
+    private static final long serialVersionUID = -8734479943734920904L;
+
+    public OAuthAPI(Main main) {
+        super(main, RECIPE_ID.OAUTH.toString());
+    }
+
+    @Override
+    public String getPath() {
+        return "oauth/auth";
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // TODO Work in progress!
+
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        String clientId = InputParser.parseStringOrThrowError(input, "clientId", false);
+        String redirectUri = InputParser.parseStringOrThrowError(input, "redirectUri", false);
+        String responseType = InputParser.parseStringOrThrowError(input, "responseType", false);
+        String scope = InputParser.parseStringOrThrowError(input, "scope", false);
+        String state = InputParser.parseStringOrThrowError(input, "state", false);
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = getTenantStorage(req);
+
+            OAuthAuthResponse authResponse = OAuth.getAuthorizationUrl(super.main, appIdentifier, storage,
+                    clientId, redirectUri, responseType, scope, state);
+            JsonObject response = new JsonObject();
+            response.addProperty("redirectTo", authResponse.redirectTo);
+
+            if (authResponse.cookies != null) {
+                Gson gson = new Gson();
```

Comment:
We use JSONObject instead of Gson. https://github.com/supertokens/supertokens-core/blob/master/src/main/java/io/supertokens/webserver/api/userroles/CreateRoleAPI.java#L91. Please follow existing patterns as much as possible

Reply:
Done